crates/abq_cli/src/abq_config.rs

```diff
@@ -0,0 +1,121 @@
+use anyhow::anyhow;
+use std::{fs, path::PathBuf};
+
+use abq_hosted::AccessToken;
+use etcetera::{app_strategy, AppStrategy, AppStrategyArgs};
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub struct AbqConfig {
+    pub rwx_access_token: AccessToken,
+}
+
+pub enum AbqConfigFileMode {
+    // Conventional unix location (e.g ~/.config/abq/config.toml)
+    Conventional,
+    // Non-standard location (e.g. ~/my/custom/path/config.toml)
+    Override(String),
+    // Ignore any located config file (For instance, if the host machine happens to have a config file)
+    Ignore,
+}
+
+pub fn abq_config_filepath(mode: AbqConfigFileMode) -> Option<PathBuf> {
+    match mode {
+        AbqConfigFileMode::Ignore => None,
+        AbqConfigFileMode::Override(path) => Some(PathBuf::from(path)),
+        AbqConfigFileMode::Conventional => {
+            let strategy = app_strategy::Unix::new(AppStrategyArgs {
+                top_level_domain: "org".to_string(),
+                author: "rwx".to_string(),
+                app_name: "abq".to_string(),
+            });
+            let config_dir =
+                Result::expect(strategy, "failed to build conventional abq config filepath")
+                    .config_dir();
+            Some(config_dir.join("config.toml"))
+        }
+    }
+}
+
+pub fn write_abq_config(
+    config: AbqConfig,
+    config_path: anyhow::Result<PathBuf>,
+) -> anyhow::Result<PathBuf> {
+    let abq_config_filepath = config_path?;
+    let config_dir = abq_config_filepath
+        .parent()
+        .ok_or_else(|| anyhow!("abq config file must have parent dir"))?;
+    fs::create_dir_all(config_dir)?;
+    let toml_str = toml::to_string(&config)?;
+    fs::write(&abq_config_filepath, toml_str)?;
+    Ok(abq_config_filepath)
+}
+
+pub fn read_abq_config(path: Option<PathBuf>) -> Option<AbqConfig> {
+    match path {
+        Some(config_path) => {
+            let toml_str = fs::read_to_string(config_path);
+            let config = toml::from_str(&toml_str.ok()?).ok();
+            Some(config?)
+        }
+        None => None,
+    }
+}
```

Comment:
```suggestion
pub fn read_abq_config(path: Option<PathBuf>) -> Option<AbqConfig> {
    let path = path?;
    let toml_str = fs::read_to_string(config_path).ok()?;
    toml::from_str(&toml_str).ok()
}
```

just a little easier to read I think :)